### ISSUE TITLE
Hardcoded WiFi is awkward when moving locations

### ISSUE THREAD

**Issue by Snowflake6** [wizche/m5paper-weather#3]
I've been carrying my M5Paper from home to work and still wanted my weather to update but didn't want to have to re-compile twice a day to get it to update.  So I modified the code to use WiFiMulti.h.

1. Take the WIFI #defines out of Config.h
2. Replace EPDWifi.h with this:
````
/*
   Copyright (C) 2021 SFini

   This program is free software: you can redistribute it and/or modify
   it under the terms of the GNU General Public License as published by
   the Free Software Foundation, either version 3 of the License, or
   (at your option) any later version.

   This program is distributed in the hope that it will be useful,
   but WITHOUT ANY WARRANTY; without even the implied warranty of
   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
   GNU General Public License for more details.

   You should have received a copy of the GNU General Public License
   along with this program.  If not, see <http://www.gnu.org/licenses/>.
*/
/**
  * @file EPDWifi.h
  * 
  * Helper functions for the wifi connection.
  */
#pragma once
#include <WiFi.h>
#include <WiFiMulti.h>

/* Start and connect to the wifi */
bool StartWiFi(int &rssi) 
{
   IPAddress dns(8, 8, 8, 8); // Google DNS

   WiFiMulti wifiMulti;

   WiFi.mode(WIFI_STA);
   WiFi.disconnect();
   WiFi.setAutoConnect(true);
   WiFi.setAutoReconnect(true);

/* add AP's here with AP Name, AP Password pairs */
   wifiMulti.addAP("WiFiAP_1", "WiFiAP_PW_1");
   wifiMulti.addAP("WiFiAP_2", "WiFiAP_PW_2");
   wifiMulti.addAP("WiFiAP_3", "WiFiAP_PW_3");
/* etc. etc. */

   Serial.println("Connecting WiFi...");
   if(wifiMulti.run() == WL_CONNECTED) {
      Serial.print("Connected to ");
      Serial.println(WiFi.SSID());
   }

   for (int retry = 0; WiFi.status() != WL_CONNECTED && retry < 30; retry++) {
      delay(500);
      Serial.print(".");
   }

   rssi = 0;
   
   if (WiFi.status() == WL_CONNECTED) {
      rssi = WiFi.RSSI();
      Serial.println("WiFi connected at: " + WiFi.localIP().toString());
      return true;
   } else {
      Serial.println("WiFi connection *** FAILED ***");
      return false;
   }
}

/* Stop the wifi connection */
void StopWiFi() 
{
   Serial.println("Stop WiFi");
   WiFi.disconnect();
   WiFi.mode(WIFI_OFF);
}
````
3. I went one further and put the AP name on the display, by making this change in Display.h:
````
void WeatherDisplay::DrawHead()
{
   canvas.drawString("AP:" + WiFi.SSID(), 20, 10);
   canvas.drawCentreString(CITY_NAME, maxX / 2, 10, 1);
   canvas.drawString(WifiGetRssiAsQuality(myData.wifiRSSI) + "%", maxX - 200, 10);
   DrawRSSI(maxX - 155, 25);
   canvas.drawString(String(myData.batteryCapacity) + "%", maxX - 110, 10);
   DrawBattery(maxX - 65, 10);
}
````

**Comment by wizche** [comment]
Hey @Snowflake6, thanks a lot! Would you mind sending a pull request for the change?

**Comment by Snowflake6** [comment]
@wizche I'd love to, but I am already operating at my limit when it comes to Github knowledge... If I knew how to do a PR, I would happily do that... :)

**Comment by Snowflake6** [comment]
@wizche Okay, I *barely* know what i'm doing but I think I managed to figure this out enough to submit a pull request... :)

**Comment by Snowflake6** [comment]
Fixed with the incorporation of pull request #4.  https://github.com/wizche/m5paper-weather/pull/4
